plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,421 @@
+
+	var fields = make(map[string]interface{})
+
+	if fullstat && first == "events" && len(nline) >= len(eventsFields) {
+		for i, t := range eventsFields {
+			fields[t] = nline[i]
+		}
+		acc.AddFields("nfs_events", fields, tags)
+	} else if fullstat && first == "bytes" && len(nline) >= len(bytesFields) {
+		for i, t := range bytesFields {
+			fields[t] = nline[i]
+		}
+		acc.AddFields("nfs_bytes", fields, tags)
+	} else if fullstat && first == "xprt" && len(line) > 1 {
+		switch line[1] {
+		case "tcp":
+			if len(nline)+2 >= len(xprttcpFields) {
+				for i, t := range xprttcpFields {
+					fields[t] = nline[i+2]
+				}
+				acc.AddFields("nfs_xprt_tcp", fields, tags)
+			}
+		case "udp":
+			if len(nline)+2 >= len(xprtudpFields) {
+				for i, t := range xprtudpFields {
+					fields[t] = nline[i+2]
+				}
+				acc.AddFields("nfs_xprt_udp", fields, tags)
+			}
+		}
+	} else if fullstat {
+		if version == "3" {
+			if nfs3Ops[first] && (len(nline) <= len(nfsopFields)) {
+				for i, t := range nline {
+					item := fmt.Sprintf("%s_%s", first, nfsopFields[i])
+					fields[item] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		} else if version == "4" {
+			if nfs4Ops[first] && (len(nline) <= len(nfsopFields)) {
+				for i, t := range nline {
+					item := fmt.Sprintf("%s_%s", first, nfsopFields[i])
+					fields[item] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		}
+	} else {
+		if first == "READ" {
+			fields["read_ops"] = nline[0]
+			fields["read_retrans"] = (nline[1] - nline[0])
+			fields["read_bytes"] = (nline[3] + nline[4])
+			fields["read_rtt"] = nline[6]
+			fields["read_exe"] = nline[7]
+			acc.AddFields("nfsstat_read", fields, tags)
+		} else if first == "WRITE" {
+			fields["write_ops"] = nline[0]
+			fields["write_retrans"] = (nline[1] - nline[0])
+			fields["write_bytes"] = (nline[3] + nline[4])
+			fields["write_rtt"] = nline[6]
+			fields["write_exe"] = nline[7]
+			acc.AddFields("nfsstat_write", fields, tags)
+		}
+	}
+
+	return nil
+}
+
+func (n *NFSClient) processText(scanner *bufio.Scanner, acc telegraf.Accumulator) error {
+	var device string
+	var version string
+	var export string
+	var skip bool
+	var nfs3Ops map[string]bool
+	var nfs4Ops map[string]bool
+	nfs3Ops = make(map[string]bool)
+	nfs4Ops = make(map[string]bool)
+
+	if len(n.IncludeOperations) == 0 {
+		for _, Op := range nfs3Fields {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range nfs4Fields {
+			nfs4Ops[Op] = true
+		}
+	} else {
+		for _, Op := range n.IncludeOperations {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range n.IncludeOperations {
+			nfs4Ops[Op] = true
+		}
+	}
+
+	if len(n.ExcludeOperations) > 0 {
+		for _, Op := range n.ExcludeOperations {
+			if nfs3Ops[Op] {
+				delete(nfs3Ops, Op)
+			}
+			if nfs4Ops[Op] {
+				delete(nfs4Ops, Op)
+			}
+		}
+	}
+
+	for scanner.Scan() {
+		line := strings.Fields(scanner.Text())
+
+		if in(line, "fstype") && (in(line, "nfs") || in(line, "nfs4")) && len(line) > 4 {
```

Comment:
Wouldn't it be easier to use a regexp for checking the line matches _before_ splitting it?

Reply:
Still open.

Reply:
Futhermore, why is `in(line, "fstype")` omitted in the `else if`?

And what happens if the line does not match any of the specified conditions? In this case `device`, `export`, etc. are kept from the previous mount and the processing probably outputs some strange metrics!
Wouldn't it be better to do an `else` branch that just continues the loop skipping the whole processing?